src/OrchardCore/OrchardCore.Queries.Core/Services/QueryManagerSession.cs

```diff
@@ -0,0 +1,39 @@
+using System.Threading.Tasks;
+using Microsoft.Extensions.Caching.Distributed;
+using OrchardCore.Environment.Shell;
+
+namespace OrchardCore.Queries.Core.Services;
+
+public sealed class QueryManagerSession
```

Comment:
Since this is only for GraphQL:

```suggestion
public sealed class QueryManagerGraphQLSession
```

Reply:
Nah. I don't like to add GraphQL into the name. It's related to the manager not GraphQL even when the end value is used mainly by GraphQl

Reply:
The name suggests, and this confused me when I first have seen this, that this is some kind of session like for `ContentManager`, or something related to user sessions, but it's neither.

Reply:
Maybe we can just remove this extra layer of abstraction and move the code directly into the `QueryManager`.

Reply:
That'd work too.

Reply:
removed it and placed it into the manager directly as suggested by Georg.